### DISCUSSION TITLE
Please show me your notification daemons!

### DISCUSSION THREAD

**Discussion by The-Compiler** [qutebrowser/qutebrowser#6326] [General] [answered]
I'm currently working on [integrating native notifications](https://github.com/qutebrowser/qutebrowser/pull/5457#issuecomment-806036584) into qutebrowser - I mainly use [Dunst](https://dunst-project.org/) myself and did a fair bit of testing, but it'd be good to see what people actually use.

Thus, could you please show the output of:

`dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation`

on your system?

Also, if you don't mind playing guinea pig and have the git repo lying around, you could do:

- `git checkout dev`
- Potentially `git fetch` and `git reset --hard origin/dev` since I force-push `dev` regularly
- `cd tests/end2end` and `python3 -m http.server`
- In a second terminal, in the repo root, `python3 -m qutebrowser --temp-basedir  localhost:8000/data/javascript/notifications.html`
- Play around with the buttons a bit, make sure the image colors look like intended, and perhaps also look at websites displaying notifications

Thanks! There's still some work to do, but I'll probably merge it to `master` in the next couple of days.

**Comment by max-baz** [comment]
```
$ dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation
method return time=1616715072.476737 sender=:1.272 -> destination=:1.372 serial=249 reply_serial=2
   string "mako"
   string "emersion"
   string "0.0.0"
   string "1.2"
```

It seems to work really well except for the "Show multiple notifications" - that button produces just one notification, with `i=3` inside. FWIW Chromium and Firefox do exactly the same for me, so it might be a mako issue 🤷‍♂️

**Reply by The-Compiler** [reply]
That's the intended behavior - I've renamed the button to make this clearer.

**Comment by toofar** [comment]
5.14.2 and 5.15.2, after saying 'n' to a notification permission request there doesn't seem any way to let the page re-request, not sure if that is intended. Tested with your instructions and https://web-push-book.gauntface.com/demos/notification-examples/ and everything is working fine (awesome doesn't support actions, sounds, delays, amending etc) apart from:
5.15.2, as with @maximbaz the "show multiple notifications" only shows the last one. On 5.14.2 it shows all three, so maybe not a mako issue. Although firefox also only shows the last one for me. Ah, it isn't supposed to show all three if you open them that fast https://developer.mozilla.org/en-US/docs/Web/API/Notifications_API/Using_the_Notifications_API#replacing_existing_notifications Adding a delay in between makes them visibly get replaced properly (on 5.15) like that describes, same with firefox.
```
$ dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation
method return time=1616743040.169931 sender=:1.7 -> destination=:1.19867 serial=60 reply_serial=2
   string "naughty"
   string "awesome"
   string "v4.3"
   string "1.0"
```
My notification thing has been a bit flaky lately so looking forward to a better one ;)

**Comment by carlosgarciafm** [comment]
```
$ dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation
method return time=1616745633.475323 sender=:1.6 -> destination=:1.9030 serial=333 reply_serial=2
   string "dunst"
   string "knopwob"
   string "1.5.0 (2020-07-23)"
   string "1.2"
```
Notifications are working fine except for the "show multiple notifications" one, where the notification is being replaced showing mostly `i=3`. It is possible to see the notification being replaced from `i=1` -> `i=3`; if you have sounds enabled for the notifications it is the more obvious.

**Reply by The-Compiler** [reply]
That's the intended behavior - I've renamed the button to make this clearer.

**Comment by The-Compiler** [comment, marked as answer]
Thanks everyone! Notification support has been merged now, and seems to work fine everywhere from what I can tell.

**Comment by Gochkin** [comment]
Don't know if it is still relevant, since the notification support has been merged. But stil, here is the output from my machine:
```
$ dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation
method return time=1618759926.352730 sender=:1.0 -> destination=:1.53 serial=5 reply_serial=2
   string "mako"
   string "emersion"
   string "0.0.0"
   string "1.2"
```

**Comment by bodograumann** [comment]
Same. This is in awesome wm:
```
> dbus-send --session --print-reply --dest=org.freedesktop.Notifications /org/freedesktop/Notifications org.freedesktop.Notifications.GetServerInformation
method return time=1618769147.416808 sender=:1.2 -> destination=:1.1373 serial=205 reply_serial=2
   string "naughty"
   string "awesome"
   string "v4.3"
   string "1.0"
```
